src/python/pants/core/util_rules/adhoc_process_support.py

```diff
@@ -657,6 +614,57 @@ async def prepare_adhoc_process(
     return _output_at_build_root(proc, bash)
 
 
+async def prepare_env_vars(
+    env_vars_templates: tuple[str, ...],
+    *,
+    extra_paths: tuple[str, ...] = (),
+    description_of_origin: str,
+) -> FrozenDict[str, str]:
+    to_fetch: set[str] = set()
+    duplicate_keys: set[str] = set()
+    env_vars: dict[str, str] = {}
+    for env_var in env_vars_templates:
```

Comment:
if you put `env_vars` in the signature instead, that would simplify the call sites using this method, not having to jump through more hopes eq. to `FrozenDict(env_vars.update(prepared_env_vars_result))`.

```diff
 async def prepare_env_vars(
     env_vars_templates: tuple[str, ...],
     *,
+    env_vars: dict[str, str] = {},
     extra_paths: tuple[str, ...] = (),
     description_of_origin: str,
 ) -> FrozenDict[str, str]:
     to_fetch: set[str] = set()
     duplicate_keys: set[str] = set()
-    env_vars: dict[str, str] = {}
     for env_var in env_vars_templates:
```

Reply:
Good idea. Done.